app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def widget_params
+    # read all json schemas
+    jsons = Dir[Rails.root.join('db','json','*.json_schema')].map{ |f| JSON.parse File.read(f) }
+    allowed_keys = []
+
+    # add allowed keys as fields, dictionaries, or hashes depending on their type
+    jsons.map{|j| j['properties'] }.each do |properties|
```

Comment:
Sounds good!

Like I said, I don't think we need to cover those edge cases specifically, I just wanted to make sure that we'd thought about them and are clear on which cases we're covering for our own future sanity.